src/python_bridge.rs

```diff
@@ -0,0 +1,177 @@
+use pyo3::exceptions::{PyIOError, PyKeyError, PyValueError};
+use pyo3::prelude::*;
+use serde_json;
+use std::collections::HashMap;
+use std::sync::{Arc, RwLock};
+
+use crate::{AmpIndexer, AmpResult, BlartAmpIndex, OriginalAmp};
+
+#[pyclass]
+#[derive(Clone)]
+pub struct PyAmpResult {
+    #[pyo3(get)]
+    pub title: String,
+    #[pyo3(get)]
+    pub url: String,
+    #[pyo3(get)]
+    pub click_url: String,
+    #[pyo3(get)]
+    pub impression_url: String,
+    #[pyo3(get)]
+    pub advertiser: String,
+    #[pyo3(get)]
+    pub block_id: i32,
+    #[pyo3(get)]
+    pub iab_category: String,
+    #[pyo3(get)]
+    pub icon: String,
+    #[pyo3(get)]
+    pub full_keyword: String,
+}
+
+impl From<AmpResult> for PyAmpResult {
+    fn from(result: AmpResult) -> Self {
+        PyAmpResult {
+            title: result.title,
+            url: result.url,
+            click_url: result.click_url,
+            impression_url: result.impression_url,
+            advertiser: result.advertiser,
+            block_id: result.block_id,
+            iab_category: result.iab_category,
+            icon: result.icon,
+            full_keyword: result.full_keyword,
+        }
+    }
+}
+
+// Thread-safe index wrapper
+type IndexHandle = Arc<RwLock<BlartAmpIndex>>;
+
+// Async index manager for FastAPI
+#[pyclass]
+pub struct AmpIndexManager {
+    indexes: Arc<RwLock<HashMap<String, IndexHandle>>>,
+}
+
+#[pymethods]
+impl AmpIndexManager {
+    #[new]
+    fn new() -> PyResult<Self> {
+        Ok(AmpIndexManager {
+            indexes: Arc::new(RwLock::new(HashMap::new())),
+        })
+    }
+
+    /// Build index from JSON file (async)
+    fn build_from_file<'py>(
```

Comment:
heh, +1 for descriptive lifetimes

Reply:
I let the PyO3 docs take credit for this :) 
https://pyo3.rs/v0.25.0/function.html